src/components/app/app-card.js

```diff
@@ -0,0 +1,221 @@
+  const [flipped, setFlipped] = useState(false)
+  const [currentMemory, setMemory] = useState(validateLocalstorageValue('memory', app_id, toBytes(minimum_resources.memory), toBytes(maximum_resources.memory)));
+  const [currentCpu, setCpu] = useState(validateLocalstorageValue('cpu', app_id, minimum_resources.cpus, maximum_resources.cpus));
+  const [currentGpu, setGpu] = useState(validateLocalstorageValue('gpu', app_id, minimum_resources.gpus, maximum_resources.gpus));
+
+  const toggleConfig = event => setFlipped(!flipped)
+
+  //app can be launched here using axios to hit the /start endpoint
+  const appLauncher = () => {
+    launchApp(app_id, currentCpu, currentGpu, formatBytes(currentMemory))
+      .then(res => {
+        addNotification({ type: 'success', text: 'Launching app successful.' })
+      }).catch(e => {
+        addNotification({ type: 'error', text: 'Error occurs when launching apps. Please try again!' })
+      })
+    localStorage.setItem(`${app_id}-cpu`, currentCpu);
+    localStorage.setItem(`${app_id}-gpu`, currentGpu);
+    localStorage.setItem(`${app_id}-memory`, currentMemory);
+  }
+
+  const getLogoUrl = (app_id) => {
+    return `https://github.com/helxplatform/app-support-prototype/raw/master/dockstore-yaml-proposals/${app_id}/icon.png`
+  }
+
+  return (
+    <Card style={{ minHeight: '350px' }}>
+      <Card.Header><AppHeader><b>{name} {flipped ? "- App Config" : <span />}</b></AppHeader></Card.Header>
+      <Relative>
+        <Card.Body>
+          <AppLogo src={'' + getLogoUrl(app_id)} />
+          <AppInfo>
+            <Paragraph>{description}</Paragraph>
+            <Paragraph dense>{detail}</Paragraph>
+            <Link to={docs}>About {name}</Link>
+          </AppInfo>
+        </Card.Body>
+        <ConfigSlider visible={flipped}>
+          <div className="specs">
+            <SpecContainer><SpecName>CPU</SpecName><SliderMinMaxContainer><SpecMinMax>Min: {minimum_resources.cpus} {minimum_resources.cpus > 1 ? 'Cores' : 'Core'}</SpecMinMax><Spec><b>{currentCpu}</b><Slider type="range" min={minimum_resources.cpus} max={maximum_resources.cpus} step="1" value={currentCpu} onChange={(e) => setCpu(e.target.value)} /></Spec><SpecMinMax>Max: {maximum_resources.cpus} {maximum_resources.cpus > 1 ? 'Cores' : 'Core'}</SpecMinMax></SliderMinMaxContainer></SpecContainer>
+            <SpecContainer><SpecName>GPU</SpecName>{minimum_resources.gpus === maximum_resources.gpus ? <SpecDefaultText>Default Setting: {minimum_resources.gpus} Core</SpecDefaultText> : <SliderMinMaxContainer><SpecMinMax>Min: {minimum_resources.gpus} {minimum_resources.gpus > 1 ? 'Cores' : 'Core'}</SpecMinMax><Spec><b>{currentGpu}</b><Slider type="range" min={minimum_resources.gpus} max={maximum_resources.gpus} step="1" value={currentGpu} onChange={(e) => setGpu(e.target.value)} /></Spec><SpecMinMax>Max: {maximum_resources.gpus} {maximum_resources.gpus > 1 ? 'Cores' : 'Core'}</SpecMinMax></SliderMinMaxContainer>}</SpecContainer>
```

Comment:
@connectbo a couple thoughts on this:

- Should the text stay left justified under `min` so the layout stays similar to the surrounding rows?
- Should we say `Locked Setting` instead of default? To me default is a recommended setup, but I can change it. In the scenario here we are trying to communicate that this setting cannot be changed/modified.
- The logic right now is based on counts, but I think that the logic should be that a setting is considered locked whenever min === max on that resource type. The min may be 0, but the max may be N, that isn't locked. The min could be 4000MB and the max may be 4000MB that's locked. Does that make sense?

![image](https://user-images.githubusercontent.com/2118138/115871274-53f51480-a40e-11eb-870a-48b3bc8c1be1.png)

Reply:
Those all makes sense to me. Let me poke around and would apply locking logic to cpu/memory as well.